pkg/controllermanager/controller/shoot/shoot_reference_control.go

```diff
@@ -69,31 +69,44 @@ func (c *Controller) shootReferenceUpdate(oldObj, newObj interface{}) {
 }
 
 func refChange(oldShoot, newShoot *gardencorev1beta1.Shoot) bool {
+	return shootDNSFieldChanged(oldShoot, newShoot) || shootKubeAPIServerAuditConfigFieldChanged(oldShoot, newShoot)
+}
+
+func shootDNSFieldChanged(oldShoot, newShoot *gardencorev1beta1.Shoot) bool {
 	return !apiequality.Semantic.Equalities.DeepEqual(oldShoot.Spec.DNS, newShoot.Spec.DNS)
 }
 
+func shootKubeAPIServerAuditConfigFieldChanged(oldShoot, newShoot *gardencorev1beta1.Shoot) bool {
+	return !apiequality.Semantic.Equalities.DeepEqual(oldShoot.Spec.Kubernetes.KubeAPIServer.AuditConfig, newShoot.Spec.Kubernetes.KubeAPIServer.AuditConfig)
```

Comment:
If `.Spec.Kubernetes.KubeAPIServer` is nil, this is panic with segmentation fault.
```suggestion
	return !apiequality.Semantic.Equalities.DeepEqual(getAuditPolicyConfigMapRef(oldShoot.Spec.Kubernetes.KubeAPIServer), getAuditPolicyConfigMapRef(newShoot.Spec.Kubernetes.KubeAPIServer))
```

Reply:
Hm, kubernetes.kubeAPIServer is always defaulted. Ref https://github.com/gardener/gardener/blob/v1.12.1/pkg/apis/core/v1beta1/defaults.go#L138-L155
I think it should not be a problem.

Reply:
Currently it will not be a problem, but if the default is changed after some time (several years, or once Gardener support only k8s clusters >=1.16), this code might not get adapted and it will become a bug.

Reply:
I get your point but it is something that needs to be adapted if (or when) the defaulting logic is changed. In a similar way in the func above `shootDNSFieldChanged` we rely that `ShootDNS` admission plugin (something that you can disable with the `--disable-admission-plugins` flag of the gardener apiserver) will default the dns section if ti is not specified.

Reply:
+1
Reasonable point, but we rely on defaults at many points in g/g.